test/e2e/service_options_test.go

```diff
@@ -29,62 +28,55 @@ func TestServiceOptions(t *testing.T) {
 	test.Setup(t)
 	defer test.Teardown(t)
 
-	t.Run("create hello service with concurrency options and returns no error", func(t *testing.T) {
-		test.serviceCreateWithOptions(t, "hello", []string{"--concurrency-limit", "250", "--concurrency-target", "300"})
+	t.Run("create and validate service with concurrency options", func(t *testing.T) {
+		test.serviceCreateWithOptions(t, "svc1", []string{"--concurrency-limit", "250", "--concurrency-target", "300"})
+		test.validateServiceConcurrencyTarget(t, "svc1", "300")
+		test.validateServiceConcurrencyLimit(t, "svc1", "250")
 	})
 
-	t.Run("returns valid concurrency options for hello service", func(t *testing.T) {
-		test.serviceDescribeConcurrencyLimit(t, "hello", "250")
-		test.serviceDescribeConcurrencyTarget(t, "hello", "300")
+	t.Run("update and validate service with concurrency limit", func(t *testing.T) {
+		test.serviceUpdate(t, "svc1", []string{"--concurrency-limit", "300"})
+		test.validateServiceConcurrencyLimit(t, "svc1", "300")
 	})
 
-	t.Run("update concurrency limit for hello service and returns no error", func(t *testing.T) {
-		test.serviceUpdate(t, "hello", []string{"--concurrency-limit", "300"})
+	t.Run("update concurrency options with invalid values for service", func(t *testing.T) {
+		test.serviceUpdateWithInvalidValue(t, "svc1", []string{"--concurrency-limit", "-1", "--concurrency-target", "0"})
```

Comment:
The one thing I don’t like here is that the method will test an invalid case in two steps. Perhaps better is to have helper method that does service update and returns error that in this “test” is validated to make sure failure is correct. Or implement all here. The reason for the helpers is that they are reusable but one that tests an invalid condition is likely not reusable IMO. What do you think?

Reply:
Ok I see. Although the other tests here uses the helper validation methods to validate particular field of service, but they are used more than once. This particular test and its helper method is checking for a particular error.
As we move along, we'll need to bring more of such negative test cases, but I am fine with implementing that two liner helper method here, we can always have it separated into helper if more tests require it.